### ISSUE TITLE
Czech Republic 15 minute data

### ISSUE THREAD

**Issue by systemcatch** [electricitymaps/electricitymaps-contrib#1899]
There is 15min generation data available at https://www.ceps.cz/en/all-data#Generation. This is better than our current data which is hourly from ENTSOE, which is lagging right now.

It's possible to go back to previous days, I also see planned net generation and hourly exchanges.

edit: possible api info at https://www.ceps.cz/en/web-services

**Comment by gecoffman** [comment]
Hello - I would like to take a crack at building a parser/creating a wrapper for the API

**Comment by gecoffman** [comment]
Looks like the API requires using SOAP - will take me a little longer, will try to have done by the end of this week.

**Comment by systemcatch** [comment]
Good stuff @gecoffman, post here if you need any advice on writing the parser.

**Comment by alixunderplatz** [comment]
I tried to have a look at the SOAP request stuff required for the CEPS datasource for CZ posted above by @systemcatch

I'm stuck at getting data back out of the request (ending up with "400 Bad Request").

It should be possible to obtain the data with "requests".
https://stackoverflow.com/questions/15569330/making-a-soap-request-using-python-requests-module

I'm sure one of you guys will be able to get the xml-data back ;) So here are a few links for proper configuration:

website with graphs for generation, power flows etc as reference:
https://www.ceps.cz/en/all-data#Generation

documentation:
https://www.ceps.cz/en/web-services (download the .docx documentation)
https://www.ceps.cz/_layouts/CepsData.asmx

For generation by type (hourly/quarterhourly):
https://www.ceps.cz/_layouts/CepsData.asmx?op=Generation
For cross-border power flows (hourly/quarterhourly/1-min):
https://www.ceps.cz/_layouts/CepsData.asmx?op=CrossborderPowerFlows
For solar+wind (hourly/quarterhourly/1-min)
https://www.ceps.cz/_layouts/CepsData.asmx?op=GenerationRES
WSDL-file:
https://www.ceps.cz/_layouts/CepsData.asmx?WSDL

Here is a snippet of the "body" of my desired sample-data if you wanna try fixing everything around it (QH=quarterhourly, AVG=average, RT=realtime, ALL=all generation types):
```
      <dateFrom>2020-05-05T00:00:00</dateFrom>
      <dateTo>2020-05-06T00:00:00</dateTo>
      <agregation>QH</agregation>
      <function>AVG</function>
      <version>RT</version>
      <para1>ALL</para1>
```

The good thing is: we could get the generation data for solar and wind as well as the exchanges in 1-min-steps, if desired.
Otherwise, the 15-min intervalls will do, as these are the smallest common timesteps of both generation and flows.

There are download options as txt csv and xml as well, but I haven't tried if that's the easier option.

The xml will look like this:

```
<?xml version="1.0"?>

-<root xmlns="https://www.ceps.cz/CepsData/StructuredData/1.0">


-<information>

<name>Generation</name>

<date_from>2020-05-05T00:00:00+02:00</date_from>

<date_to>2020-05-05T23:59:59+02:00</date_to>

<version>RT</version>

<function>AVG</function>

<aggregation>QH</aggregation>

<typeofgeneratingplant>all</typeofgeneratingplant>

</information>


-<series>

<serie name="TPP [MW]" id="value1"/>

<serie name="CCGT [MW]" id="value2"/>

<serie name="NPP [MW]" id="value3"/>

<serie name="HPP [MW]" id="value4"/>

<serie name="PsPP [MW]" id="value5"/>

<serie name="AltPP [MW]" id="value6"/>

<serie name="ApPP [MW]" id="value7"/>

<serie name="WPP [MW]" id="value8"/>

<serie name="PVPP [MW]" id="value9"/>

</series>


-<data>

<item value9="0" value8="120.8" value7="0" value6="204.4" value5="0" value4="93.7" value3="2195.4" value2="261.7" value1="3165.7" date="2020-05-05T00:00:00+02:00"/>

<item value9="0" value8="108.3" value7="0" value6="201.9" value5="0" value4="93.3" value3="1978.1" value2="260.4" value1="3134.5" date="2020-05-05T00:15:00+02:00"/>

<item value9="0" value8="111.5" value7="0" value6="199.4" value5="0" value4="93.3" value3="1978.1" value2="260.7" value1="3138.3" date="2020-05-05T00:30:00+02:00"/>

<item value9="0" value8="114.6" value7="0" value6="199.4" value5="0" value4="93.3" value3="1980.5" value2="258.9" value1="3132.2" date="2020-05-05T00:45:00+02:00"/>

<item value9="0" value8="109.3" value7="0" value6="198" value5="0" value4="93.3" value3="1978.1" value2="268.4" value1="3151.7" date="2020-05-05T01:00:00+02:00"/>

<item value9="0" value8="100.1" value7="0" value6="198.5" value5="0" value4="93.3" value3="1977.8" value2="277.3" value1="3129.2" date="2020-05-05T01:15:00+02:00"/>

<item value9="0" value8="100.7" value7="0" value6="198.5" value5="0" value4="93.3" value3="2075" value2="263.7" value1="3149.6" date="2020-05-05T01:30:00+02:00"/>

<item value9="0" value8="98.3" value7="0" value6="199.3" value5="0" value4="93.3" value3="3070.7" value2="259.5" value1="3122.4" date="2020-05-05T01:45:00+02:00"/>

<item value9="0" value8="95.4" value7="0" value6="198.5" value5="0" value4="93.3" value3="3070.7" value2="263.7" value1="3088.9" date="2020-05-05T02:00:00+02:00"/>

<item value9="0" value8="90.3" value7="0" value6="201.9" value5="0" value4="93.3" value3="3070.9" value2="260.6" value1="3041.8" date="2020-05-05T02:15:00+02:00"/>

<item value9="0" value8="88.5" value7="0" value6="201.3" value5="0" value4="93.3" value3="3073.8" value2="267.5" value1="3028.6" date="2020-05-05T02:30:00+02:00"/>

<item value9="0" value8="79.5" value7="0" value6="201" value5="0" value4="93.3" value3="3075.1" value2="260" value1="3002.1" date="2020-05-05T02:45:00+02:00"/>

<item value9="0" value8="73" value7="0" value6="199.7" value5="0" value4="93.3" value3="3074.4" value2="257.6" value1="2897.6" date="2020-05-05T03:00:00+02:00"/>

<item value9="0" value8="69.7" value7="0" value6="199.6" value5="0" value4="93.3" value3="3076.8" value2="262.3" value1="2792.5" date="2020-05-05T03:15:00+02:00"/>

<item value9="0" value8="68.1" value7="0" value6="198.2" value5="0" value4="93.3" value3="3076.8" value2="266.9" value1="2744.3" date="2020-05-05T03:30:00+02:00"/>

<item value9="0" value8="68.4" value7="0" value6="198.8" value5="0" value4="93.3" value3="3076.8" value2="262.2" value1="2747.3" date="2020-05-05T03:45:00+02:00"/>

<item value9="0" value8="71.8" value7="0" value6="197.6" value5="0" value4="93.3" value3="3076.8" value2="259.8" value1="2838.9" date="2020-05-05T04:00:00+02:00"/>

<item value9="0" value8="70.1" value7="0" value6="198.1" value5="0" value4="93.3" value3="3076.8" value2="259.2" value1="2874.4" date="2020-05-05T04:15:00+02:00"/>

<item value9="5.3" value8="67.7" value7="0" value6="197.6" value5="0" value4="93.3" value3="3079.1" value2="263.9" value1="2899.9" date="2020-05-05T04:30:00+02:00"/>

<item value9="8.8" value8="77.2" value7="0" value6="197.6" value5="0" value4="93.3" value3="3080.2" value2="275.9" value1="2952.1" date="2020-05-05T04:45:00+02:00"/>

<item value9="8.8" value8="81.3" value7="0" value6="197.8" value5="0" value4="93.3" value3="3083.7" value2="323.1" value1="2974.3" date="2020-05-05T05:00:00+02:00"/>

<item value9="8.8" value8="82.2" value7="0" value6="199.2" value5="0" value4="93.3" value3="3084.2" value2="426.4" value1="2923.7" date="2020-05-05T05:15:00+02:00"/>

<item value9="8.8" value8="73.2" value7="0" value6="199.9" value5="0" value4="93.3" value3="3082.7" value2="562.8" value1="2917" date="2020-05-05T05:30:00+02:00"/>

<item value9="11.4" value8="75.3" value7="0" value6="204.1" value5="29.8" value4="93.3" value3="3082.2" value2="659.3" value1="2906.3" date="2020-05-05T05:45:00+02:00"/>

<item value9="24.7" value8="73" value7="0" value6="238.4" value5="236.9" value4="114" value3="3084.6" value2="883" value1="2831.8" date="2020-05-05T06:00:00+02:00"/>

<item value9="45.4" value8="61.9" value7="0" value6="251.8" value5="210.3" value4="116.7" value3="3084.6" value2="1050.5" value1="2930.5" date="2020-05-05T06:15:00+02:00"/>

<item value9="74.5" value8="58.9" value7="0" value6="256.4" value5="208.1" value4="116.7" value3="3084.6" value2="1085.3" value1="2981" date="2020-05-05T06:30:00+02:00"/>

<item value9="113.5" value8="59.6" value7="0" value6="257.4" value5="226.3" value4="116.7" value3="3084.6" value2="1087.9" value1="3173" date="2020-05-05T06:45:00+02:00"/>

<item value9="167.3" value8="57.7" value7="0" value6="261.1" value5="183.6" value4="200" value3="3087.2" value2="1074.5" value1="3070.5" date="2020-05-05T07:00:00+02:00"/>

<item value9="238.9" value8="49.4" value7="0" value6="261.6" value5="196.5" value4="212.1" value3="3086.1" value2="1088.1" value1="3124.2" date="2020-05-05T07:15:00+02:00"/>

<item value9="325.6" value8="42.8" value7="0" value6="261.3" value5="145.9" value4="212.1" value3="3084.3" value2="1083.5" value1="3124.7" date="2020-05-05T07:30:00+02:00"/>

<item value9="420" value8="42.9" value7="0" value6="260.7" value5="159.2" value4="211.4" value3="3082.6" value2="1080.6" value1="3133.8" date="2020-05-05T07:45:00+02:00"/>

<item value9="514.6" value8="39.3" value7="0" value6="257.3" value5="203.5" value4="167.9" value3="3082.6" value2="1071.6" value1="3128.8" date="2020-05-05T08:00:00+02:00"/>

<item value9="599.4" value8="38.7" value7="0" value6="254" value5="205.5" value4="168.3" value3="3082.6" value2="1074.2" value1="3125.7" date="2020-05-05T08:15:00+02:00"/>

<item value9="690.8" value8="35.9" value7="0" value6="249.1" value5="187.7" value4="168.3" value3="3082.6" value2="1077.3" value1="3147.1" date="2020-05-05T08:30:00+02:00"/>

<item value9="792.8" value8="31.2" value7="0" value6="252.1" value5="174" value4="168.3" value3="3082.6" value2="1072.7" value1="3156.5" date="2020-05-05T08:45:00+02:00"/>

<item value9="846.3" value8="36.9" value7="0" value6="235.5" value5="0" value4="103.4" value3="3083.5" value2="1080.2" value1="3243.8" date="2020-05-05T09:00:00+02:00"/>

<item value9="904.6" value8="36.2" value7="0" value6="234.2" value5="0" value4="99.5" value3="3085.8" value2="1077.8" value1="3234.6" date="2020-05-05T09:15:00+02:00"/>

<item value9="974.4" value8="39" value7="0" value6="237" value5="0" value4="99.5" value3="3085.8" value2="1074.8" value1="3210.1" date="2020-05-05T09:30:00+02:00"/>

<item value9="1005.6" value8="38.9" value7="0" value6="238.7" value5="0" value4="99.5" value3="3085.8" value2="1076.6" value1="3199.3" date="2020-05-05T09:45:00+02:00"/>

<item value9="1039.6" value8="33.8" value7="0" value6="232.2" value5="0" value4="113.7" value3="3085.8" value2="1070.6" value1="3321.4" date="2020-05-05T10:00:00+02:00"/>

<item value9="1097.4" value8="36" value7="0" value6="233.7" value5="0" value4="105.3" value3="3085" value2="1078.8" value1="3338" date="2020-05-05T10:15:00+02:00"/>

<item value9="1159.9" value8="29.4" value7="0" value6="234.5" value5="0" value4="94.7" value3="3082.8" value2="1080.2" value1="3320" date="2020-05-05T10:30:00+02:00"/>

<item value9="1184.1" value8="31.9" value7="0" value6="236.6" value5="0" value4="98.3" value3="3085.8" value2="1077.5" value1="3316.3" date="2020-05-05T10:45:00+02:00"/>

<item value9="1224.2" value8="39.4" value7="0" value6="232.4" value5="0" value4="91.4" value3="3085.8" value2="1064.7" value1="3246.5" date="2020-05-05T11:00:00+02:00"/>

<item value9="1207.6" value8="50.8" value7="0" value6="226.7" value5="0" value4="91.4" value3="3085.8" value2="1036.3" value1="3234.9" date="2020-05-05T11:15:00+02:00"/>

<item value9="1194.9" value8="50.6" value7="0" value6="220.1" value5="0" value4="91.4" value3="3085.8" value2="705.8" value1="3302.3" date="2020-05-05T11:30:00+02:00"/>

<item value9="1215.7" value8="52.6" value7="0" value6="221.2" value5="0" value4="91.4" value3="3084.4" value2="365.8" value1="3510.2" date="2020-05-05T11:45:00+02:00"/>

<item value9="1182.4" value8="58" value7="0" value6="213.9" value5="75.1" value4="92.8" value3="3085.7" value2="271.1" value1="3291" date="2020-05-05T12:00:00+02:00"/>

<item value9="1182.9" value8="63.2" value7="0" value6="211.2" value5="70.2" value4="91.5" value3="3084.2" value2="264.7" value1="3230.2" date="2020-05-05T12:15:00+02:00"/>

<item value9="1182.2" value8="65.9" value7="0" value6="211.6" value5="70.2" value4="91.5" value3="3082.1" value2="262.1" value1="3193.7" date="2020-05-05T12:30:00+02:00"/>

<item value9="1164.8" value8="73.9" value7="0" value6="217.5" value5="59" value4="91.5" value3="3079.4" value2="259.3" value1="3204" date="2020-05-05T12:45:00+02:00"/>

<item value9="1136.6" value8="73.2" value7="0" value6="220.8" value5="0" value4="91.5" value3="3079.4" value2="279.2" value1="3348.2" date="2020-05-05T13:00:00+02:00"/>

<item value9="1152.6" value8="64" value7="0" value6="222.8" value5="0" value4="91.5" value3="3082.9" value2="283.4" value1="3309.1" date="2020-05-05T13:15:00+02:00"/>

<item value9="1112" value8="74" value7="0" value6="224.7" value5="0" value4="91.5" value3="3082.9" value2="273.1" value1="3259.2" date="2020-05-05T13:30:00+02:00"/>

<item value9="1110.1" value8="62.4" value7="0" value6="226.2" value5="0" value4="91.5" value3="3082.9" value2="262.3" value1="3231.4" date="2020-05-05T13:45:00+02:00"/>

<item value9="1120.7" value8="61" value7="0" value6="229.5" value5="0" value4="100.2" value3="3080.9" value2="270.7" value1="3344.6" date="2020-05-05T14:00:00+02:00"/>

<item value9="1066.7" value8="89.3" value7="0" value6="233" value5="0" value4="100.2" value3="3082" value2="269" value1="3289.6" date="2020-05-05T14:15:00+02:00"/>

<item value9="1013.4" value8="99.6" value7="0" value6="232.6" value5="0" value4="97.7" value3="3082" value2="265" value1="3285" date="2020-05-05T14:30:00+02:00"/>

<item value9="994.4" value8="95.8" value7="0" value6="229.2" value5="0" value4="97" value3="3082" value2="265" value1="3265.9" date="2020-05-05T14:45:00+02:00"/>

<item value9="974" value8="88.8" value7="0" value6="228.2" value5="0" value4="97" value3="3082" value2="265.7" value1="3251.6" date="2020-05-05T15:00:00+02:00"/>

<item value9="903.5" value8="88.7" value7="0" value6="227.5" value5="0" value4="97" value3="3079.5" value2="268.6" value1="3251.6" date="2020-05-05T15:15:00+02:00"/>

<item value9="868" value8="75.2" value7="0" value6="229.4" value5="0" value4="97" value3="3078.6" value2="266.3" value1="3254.4" date="2020-05-05T15:30:00+02:00"/>

<item value9="783.8" value8="78.4" value7="0" value6="232.7" value5="0" value4="97" value3="3078.6" value2="269" value1="3273.1" date="2020-05-05T15:45:00+02:00"/>

<item value9="759.2" value8="88.8" value7="0" value6="232.9" value5="0" value4="97" value3="3077.1" value2="330.2" value1="3329.1" date="2020-05-05T16:00:00+02:00"/>

<item value9="718.9" value8="93" value7="0" value6="234" value5="0" value4="97" value3="3075.4" value2="445.6" value1="3311.1" date="2020-05-05T16:15:00+02:00"/>

<item value9="659.5" value8="95.1" value7="0" value6="234.9" value5="0" value4="97" value3="3075.4" value2="604.3" value1="3303.8" date="2020-05-05T16:30:00+02:00"/>

<item value9="578.7" value8="94.9" value7="0" value6="236.1" value5="0" value4="97" value3="3075.4" value2="668.5" value1="3301.2" date="2020-05-05T16:45:00+02:00"/>

<item value9="523.5" value8="98.8" value7="0" value6="244.6" value5="0" value4="100.2" value3="3075.9" value2="937.8" value1="3288.5" date="2020-05-05T17:00:00+02:00"/>

<item value9="468.8" value8="108.8" value7="0" value6="247.5" value5="0" value4="100.2" value3="3075.5" value2="1072.8" value1="3271.3" date="2020-05-05T17:15:00+02:00"/>

<item value9="416.2" value8="113.5" value7="0" value6="239.4" value5="0" value4="100.2" value3="3078.2" value2="1073.7" value1="3275.1" date="2020-05-05T17:30:00+02:00"/>

<item value9="348.1" value8="118.5" value7="0" value6="240.7" value5="0" value4="100.2" value3="3078.9" value2="1077.4" value1="3303.4" date="2020-05-05T17:45:00+02:00"/>

<item value9="282.9" value8="121" value7="0" value6="254.1" value5="0" value4="112.4" value3="3081.3" value2="1076.1" value1="3296" date="2020-05-05T18:00:00+02:00"/>

<item value9="225.7" value8="110.8" value7="0" value6="256.8" value5="0" value4="112.4" value3="3082.1" value2="1079.3" value1="3326.4" date="2020-05-05T18:15:00+02:00"/>

<item value9="173" value8="108.2" value7="0" value6="256.1" value5="0" value4="114" value3="3082.1" value2="1084.1" value1="3379.9" date="2020-05-05T18:30:00+02:00"/>

<item value9="128.7" value8="110.1" value7="0" value6="257.5" value5="153.5" value4="115.5" value3="3082.1" value2="1085.5" value1="3389.1" date="2020-05-05T18:45:00+02:00"/>

<item value9="93" value8="98" value7="0" value6="257.2" value5="643.7" value4="249.4" value3="3085" value2="1069.9" value1="3322.1" date="2020-05-05T19:00:00+02:00"/>

<item value9="66.8" value8="97.3" value7="0" value6="258.7" value5="671.8" value4="271.1" value3="3085.4" value2="1071.3" value1="3315.1" date="2020-05-05T19:15:00+02:00"/>

<item value9="44.6" value8="93.2" value7="0" value6="256" value5="665.6" value4="272.5" value3="3082.4" value2="1075.2" value1="3302.4" date="2020-05-05T19:30:00+02:00"/>

<item value9="27" value8="90.5" value7="0" value6="252.7" value5="669.3" value4="273.1" value3="3082.4" value2="1078.4" value1="3302.9" date="2020-05-05T19:45:00+02:00"/>

<item value9="16.8" value8="91" value7="0" value6="250" value5="858" value4="316.6" value3="3084.1" value2="1075.5" value1="3269.4" date="2020-05-05T20:00:00+02:00"/>

<item value9="11.8" value8="87" value7="0" value6="251.7" value5="861.3" value4="317.2" value3="3085.6" value2="1071.5" value1="3249.8" date="2020-05-05T20:15:00+02:00"/>

<item value9="9.1" value8="94.9" value7="0" value6="249.7" value5="860.9" value4="314" value3="3087.3" value2="1066.9" value1="3226.5" date="2020-05-05T20:30:00+02:00"/>

<item value9="9.1" value8="100.2" value7="0" value6="247.6" value5="860.9" value4="314.1" value3="3088.7" value2="1073.4" value1="3249.4" date="2020-05-05T20:45:00+02:00"/>

<item value9="7.9" value8="102.5" value7="0" value6="227.3" value5="746.5" value4="166" value3="3088.7" value2="1099.3" value1="3424.9" date="2020-05-05T21:00:00+02:00"/>

<item value9="0" value8="106.1" value7="0" value6="224.4" value5="788.6" value4="162.9" value3="3089.1" value2="1085.7" value1="3322.6" date="2020-05-05T21:15:00+02:00"/>

<item value9="0" value8="100.4" value7="0" value6="223" value5="869.1" value4="162.9" value3="3092.1" value2="1070.7" value1="3222.9" date="2020-05-05T21:30:00+02:00"/>

<item value9="0" value8="103.6" value7="0" value6="220.3" value5="531.7" value4="162.4" value3="3092.1" value2="1071.5" value1="3225.1" date="2020-05-05T21:45:00+02:00"/>

<item value9="0" value8="110.3" value7="0" value6="202" value5="4.6" value4="104.7" value3="3092.7" value2="1083.4" value1="3294.1" date="2020-05-05T22:00:00+02:00"/>

<item value9="0" value8="116.1" value7="0" value6="205.2" value5="0" value4="103.4" value3="3095.2" value2="1077.9" value1="3272.1" date="2020-05-05T22:15:00+02:00"/>

<item value9="0" value8="119" value7="0" value6="209.8" value5="0" value4="103.4" value3="3095.2" value2="1077.9" value1="3271.6" date="2020-05-05T22:30:00+02:00"/>

<item value9="0" value8="101.5" value7="0" value6="210.7" value5="0" value4="103.4" value3="3095.2" value2="1074.4" value1="3240.3" date="2020-05-05T22:45:00+02:00"/>

<item value9="0" value8="94.1" value7="0" value6="210.8" value5="0" value4="103.4" value3="3095.2" value2="1073" value1="3332.9" date="2020-05-05T23:00:00+02:00"/>

<item value9="0" value8="88" value7="0" value6="212.3" value5="0" value4="103.4" value3="3095.2" value2="1068.3" value1="3260.8" date="2020-05-05T23:15:00+02:00"/>

<item value9="0" value8="81" value7="0" value6="210.4" value5="0" value4="103.4" value3="3095.2" value2="1069.8" value1="3247.4" date="2020-05-05T23:30:00+02:00"/>

<item value9="0" value8="77.6" value7="0" value6="206.6" value5="0" value4="103.4" value3="3097.1" value2="1034.5" value1="3120.8" date="2020-05-05T23:45:00+02:00"/>

</data>

</root>```

**Comment by systemcatch** [comment]
Hi Alex, I had a go at getting the data in txt (csv) format.

The link for that is https://www.ceps.cz/download-data/?format=txt which returns a base64 encoded string, converting to regular text I get the following.

```csv
Data version;From;To;Agregation function;Agregation;Type of power plant;
real data;10.05.2020 00:00:00;11.05.2020 23:59:59;agregation average;15 minutes;ALL;
Date;TPP [MW];CCGT [MW];NPP [MW];HPP [MW];PsPP [MW];AltPP [MW];ApPP [MW];WPP [MW];PVPP [MW];
10.05.2020 00:00;2357.1;816.3;3028.6;99.2;0;212.7;0;18.3;0;
10.05.2020 00:15;2443.5;469.2;3023.9;98.7;0;210.6;0;21.5;0;
10.05.2020 00:30;2475.3;264.5;3028.9;98.7;0;208.8;0;22;0;
10.05.2020 00:45;2434.1;259;3031.9;98.7;0;210.6;0;20.2;0;
10.05.2020 01:00;2321.4;257.9;3035.4;98.7;77.7;209.1;0;20.8;0;
10.05.2020 01:15;2303.6;256;3038.9;98.7;75.1;210.7;0;20.2;0;
...snipped
```
I will try to take a look at the xml stuff later today.

**Comment by willbeaufoy** [comment]
Looks like progress on this bug has stalled, I can try and fix it if it's still relevant?

**Comment by Kongkille** [comment]
It seems this is still a valid data source improvement. Would be great if someone wanted to take a look :)

**Comment by meme1255** [comment]
I've posted this data source to the Slack in 2019 – what's the issue with the data? I see some bug is mentioned here although I haven't found any details? Perhaps I might be able to help somehow. Thanks!

**Comment by Kongkille** [comment]
@meme1255 amazing!

I don't think there's an issue with the data. Not one that we know of yet anyways! I think it was meant it more as a general term for the issue progress being stalled

**Comment by meme1255** [comment]
[](url)

> 
I've implemented the proposed parser.
https://github.com/electricitymap/electricitymap-contrib/compare/master...meme1255:electricitymap-contrib:patch-1

**Comment by meme1255** [comment]
> @meme1255 amazing!
> 
> 
> 
> I don't think there's an issue with the data. Not one that we know of yet anyways! I think it was meant it more as a general term for the issue progress being stalled

https://github.com/electricitymap/electricitymap-contrib/pull/4480

**Comment by q--** [comment]
I think this can be closed now that #5112 is merged?

**Comment by VIKTORVAV99** [comment]
There is still a few data types we could migrate (like planned generation and consumption) but let's close this and if the need arises open up a new issue for those.

Thanks for doing some housekeeping @q--!
